### PR TITLE
Authentication for Powerpages webextension

### PR DESCRIPTION
**Pull request by vedg**
Added changes for dataverse authentication for Power pages

Requirements added here: https://github.com/microsoft/vscode-dev/issues/477

Testing:

npm run compile-web
tested with portal route - http://localhost:3000/powerplatform/default/adx_webtemplates/97a5603a-3ab0-ec11-9840-000d3a5a4b78?orgUrl=orgca400f6b.crm.dynamics.com&source=dataverse&schema=portalDataverseoldschema

### CODE REVIEW

**Comment by TylerLeonhardt** [comment]
As an aside, can you tell me why this extension needs to be activated for:

* `onStartupFinished`
* `onLanguage:yaml`
* `onLanguage:html`

From my perspective, this extension isn't going to be used all the time to warrant these events and an extension activating when it shouldn't need to ends up annoying customers and impacts performance.

File: src/web/client/extension.ts [hunk 1]
```diff
@@ -20,12 +23,46 @@ export function activate(context: vscode.ExtensionContext): void {
     context.subscriptions.push(
         vscode.commands.registerCommand(
             "microsoft-powerapps-portals.webExtension.init",
```
[the remark this answers is not in the document]
Comment (ashishchoudhary001) [reply src/web/client/extension.ts:25]: This command is executed from the VSCode.dev server after creating the workbench and is the first command that gets executed.

			onDidCreateWorkbench: {
				runCommands: [
					{ command: 'microsoft-powerapps-portals.webExtension.init', args: [{ appName, entity, entityId, searchParams }] },
				],
			},

File: src/web/client/extension.ts [hunk 2]
```diff
@@ -20,12 +23,48 @@ export function activate(context: vscode.ExtensionContext): void {
...
+                        for (const pair of queryParams.entries()) {
+                            queryParamsMap.set(pair[0], pair[1]);
+                        }
+                    }
+                }
+                catch (error) {
+                    vscode.window.showErrorMessage("Error encountered in query parameters fetch");
+                }
+                let accessToken;
+                if (appName != null && appName != undefined) {
```
Comment (ashishchoudhary001) [inline src/web/client/extension.ts:51]: same as above

File: src/web/client/extension.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply src/web/client/extension.ts:25]: So in that case, the code in this PR is _only_ run in your special vscode.dev route and not in Desktop/vscode.dev without your route?

**Pull request merged**